x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/ExpressionTests.java

```diff
@@ -0,0 +1,192 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+package org.elasticsearch.xpack.eql.parser;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Literal;
+import org.elasticsearch.xpack.ql.expression.UnresolvedAttribute;
+import org.elasticsearch.xpack.ql.expression.function.UnresolvedFunction;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.And;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.Or;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Neg;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.Equals;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.NotEquals;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+public class ExpressionTests extends ESTestCase {
+
+    private final EqlParser parser = new EqlParser();
+
+    public Expression expr(String source) {
+        return parser.createExpression(source);
+    }
+
+    public void testStrings() throws Exception {
+        assertEquals("hello\"world", AstBuilder.unquoteString("'hello\"world'"));
+        assertEquals("hello'world", AstBuilder.unquoteString("\"hello'world\""));
+        assertEquals("hello\nworld", AstBuilder.unquoteString("'hello\\nworld'"));
+        assertEquals("hello\\\nworld", AstBuilder.unquoteString("'hello\\\\\\nworld'"));
+        assertEquals("hello\\\"world", AstBuilder.unquoteString("'hello\\\\\\\"world'"));
+
+        // test for unescaped strings: ?"...." or ?'....'
+        assertEquals("hello\"world", AstBuilder.unquoteString("?'hello\"world'"));
+        assertEquals("hello\\\"world", AstBuilder.unquoteString("?'hello\\\"world'"));
+        assertEquals("hello'world", AstBuilder.unquoteString("?\"hello'world\""));
+        assertEquals("hello\\nworld", AstBuilder.unquoteString("?'hello\\nworld'"));
+        assertEquals("hello\\\\nworld", AstBuilder.unquoteString("?'hello\\\\nworld'"));
+        assertEquals("hello\\\\\\nworld", AstBuilder.unquoteString("?'hello\\\\\\nworld'"));
+        assertEquals("hello\\\\\\\"world", AstBuilder.unquoteString("?'hello\\\\\\\"world'"));
+    }
+
+    public void testLiterals() {
+        assertEquals(Literal.TRUE, parser.createExpression("true"));
```

Comment:
why not use `expr` everywhere instead of `parser.createExpression`

Reply:
definitely will switch this. at one point, parser.createExpression was too cumbersome or verbose, so I added this but didn't go back to updating the earlier tests